core/src/test/java/org/apache/iceberg/avro/TestAvroNameMapping.java

```diff
@@ -90,7 +90,7 @@ public void testMapProjections() throws IOException {
     projected = writeAndRead(writeSchema, readSchema, record, nameMapping);
     Record projectedL1 = ((Map<String, Record>) projected.get("location")).get("l1");
     Assert.assertNotNull("Field missing from table mapping is renamed", projectedL1.getSchema().getField("long_r2"));
-    Assert.assertNull("location.value.long, should not be read", projectedL1.get("long_r2"));
+    Assert.assertNull("location.value.long, should not be read", projectedL1.get("long"));
```

Comment:
Why did this need to change?

Reply:
I don't think the test worked as intended, it is perpendicular to the test the line above.

Reply:
The line above tests that the expected schema renamed the field. This checks that the renamed field does not have a value even though "long" was in the file. I think this should be changed back.

Reply:
I took a while to debug this because it was unclear where the error was. Could be both while reading, or writing the file. It is in the ResolvingDecoder/Parser. Got a test case that fails on Avro master: https://github.com/apache/avro/pull/752